pages/posts/validating-openapi-and-json-schema.md

```diff
@@ -0,0 +1,299 @@
+```
+
+That's enough to get the Schema Object validation we were after, but there are a
+few loose ends we'll want to tie up as well. The `jsonSchemaDialect` field in
+the OpenAPI 3.1 document can be used to change the dialect used. Since this
+schema only supports the default dialect, we want to restrict people from
+changing that to something else. If they need to change it, they'll need a
+different schema to validate against. We also don't want people using the
+`$schema` keyword to change the dialect of individual schemas.
+
+```yaml
+$schema: 'https://json-schema.org/draft/2020-12/schema'
+
+$ref: 'https://spec.openapis.org/oas/3.1/schema'
+properties:
+  jsonSchemaDialect:
+    $ref: '#/$defs/dialect'
+
+$defs:
+  dialect:
+    const: 'https://spec.openapis.org/oas/3.1/dialect/base'
+  schema:
+    $dynamicAnchor: meta
+    $ref: 'https://spec.openapis.org/oas/3.1/dialect/base'
+    properties:
+      $schema:
+        $ref: '#/$defs/dialect'
+```
+
+With that, we have exactly what you'll find in the official
+`https://spec.openapis.org/oas/3.1/schema-base` schema.
+
+## Supporting multiple dialects
+
+With the adoption of JSON Schema 2020-12 came support for the `$id` and
+`$schema` keywords, which together allows us to override the default JSON Schema
+dialect for a schema. Let's assume we have an OpenAPI 3.1 document where we use
+JSON Schema 2020-12 by default, but we also have some legacy JSON Schema
+draft-07 schemas that we want to use as well.
+
+```yaml
+jsonSchemaDialect: 'https://json-schema.org/draft/2020-12/schema'
+components:
+  schemas:
+    foo:
+      type: object
+      properties:
+        foo:
+          $ref: '#/components/schemas/baz'
+      unevaluatedProperties: false
+    bar:
+      $id: './schemas/bar'
+      $schema: 'http://json-schema.org/draft-07/schema#'
+      type: object
+      properties:
+        bar:
+          $ref: '#/definitions/number'
+      definitions:
+        number:
+          type: number
+    baz:
+      type: string
+```
+
+### What's going on here
+
+First, we use the `jsonSchemaDialect` field to set the default dialect for the
+document. By setting the default dialect to JSON Schema 2020-12, by default,
+schema will not understand the keywords added in the OpenAPI 3.1 vocabulary such
+as `discriminator`. Only standard JSON Schema 2020-12 keywords will be
+recognized.
+
+The `/components/schemas/foo` schema is understood to be interpreted as JSON
+Schema 2020-12 because that's what we set to be the default.
+
+The `/components/schemas/bar` schema changes the dialect of that schema to be
+JSON Schema draft-07. There are a couple of things working together to make this
+possible. The `$schema` keyword sets the dialect for the schema, but `$schema`
+is only allowed at the root of the document it appears in. That's why we also
+need to include the `$id` keyword. The `$id` keyword effectively makes that
+schema a separate document with it's own identifier and that location as the
```

Comment:
it's -> its

Reply:
Good catch. I'll make the change.

This PR was already merged, but this is still welcome. Thanks.
Feel free to make PRs if you see anything further.

Reply:
https://github.com/json-schema-org/blog/commit/3ffd604bb77c3f5fea41213be4d46c24f7df820b